### PR TITLE
Subset by level

### PR DESCRIPTION
**Pull request by agstephens**
I have implemented `subset_level` and added a set of tests to support it. Details are:
 - `clisops.core.subset::subset_level`: takes the arguments:
  - "first_level", "last_level"
 - `clisops.ops.subset::subset`: now passes content of "level" argument
   through to `core.subset::subset_level`.
 - tests written and they pass
 - calls to `subset_level` also added in to the spatial subset functions in `clisops.core.subset` - following the structure already used for `subset_time`

### CODE REVIEW

**Comment by agstephens** [comment]
@ellesmith88 @huard @Zeitsperre Please take a look at this and check that it is all in order.

**Review by Zeitsperre — APPROVED** [review]
Looking really great!

File: clisops/core/subset.py [hunk 1]
```diff
@@ -1044,6 +1170,57 @@ def subset_time(
...
+) -> Union[xarray.DataArray, xarray.Dataset]:
+    """Subset input DataArray or Dataset based on first and last levels.
+    Return a subset of a DataArray or Dataset for levels falling within the provided bounds.
+
+    Parameters
+    ----------
+    da : Union[xarray.DataArray, xarray.Dataset]
+      Input data.
+    first_level : Optional[Union[int, float]]
+      First level of the subset.
```
Comment (huard) [inline clisops/core/subset.py:1187]: I suggest to clarify that this is not the level index, but the level value.

**Review by huard — APPROVED** [review]

**Comment by agstephens** [comment]
@Zeitsperre @huard Thanks, I've updated based on your suggestions.

File: clisops/core/subset.py [hunk 2]
```diff
@@ -104,6 +107,90 @@ def func_checker(*args, **kwargs):
...
+    @wraps(func)
+    def func_checker(*args, **kwargs):
+        """Verify that first and last levels are valid in a level subsetting function."""
+        da = args[0]
+
+        level = xu.get_coord_by_type(da, "level", ignore_aux_coords=True)
+
+        if level is None:
+            raise Exception(
+                '{subset_level.__name__} requires input data a recognisable "level" coordinate.'
```
[the remark this answers is not in the document]
Comment (agstephens) [reply clisops/core/subset.py]: Fixed.

File: clisops/core/subset.py [hunk 3]
```diff
@@ -1044,6 +1170,57 @@ def subset_time(
...
+
+    Returns
+    -------
+    Union[xarray.DataArray, xarray.Dataset]
+      Subsetted xarray.DataArray or xarray.Dataset
+
+    Examples
+    --------
+    >>> import xarray as xr  # doctest: +SKIP
+    >>> from xclim.subset import subset_time  # doctest: +SKIP
```
[the remark this answers is not in the document]
Comment (agstephens) [reply clisops/core/subset.py]: Fixed all references to `xclim.subset` in doc strings.

File: clisops/core/subset.py (hunk 1 above)
Comment (agstephens) [reply clisops/core/subset.py]: I have updated the doc string to make it clearer.

**Pull request merged**